src/Adapters/AuthorityBasedApprovalAuthorizer.php

```diff
@@ -4,20 +4,39 @@
 
 namespace ProfessionalWiki\PageApprovals\Adapters;
 
-use MediaWiki\Page\PageIdentity;
+use Iterator;
 use MediaWiki\Permissions\Authority;
 use ProfessionalWiki\PageApprovals\Application\ApprovalAuthorizer;
+use ProfessionalWiki\PageApprovals\Application\ApproverRepository;
+use Title;
+use WikiPage;
 
 class AuthorityBasedApprovalAuthorizer implements ApprovalAuthorizer {
 
 	public function __construct(
-		private Authority $authority
+		private Authority $authority,
+		private ApproverRepository $approverRepository
 	) {
 	}
 
-	public function canApprove( PageIdentity $page ): bool {
-		// TODO: logic
-		return true;
+	public function canApprove( WikiPage $page ): bool {
+		$sharedCategories = array_intersect(
+			$this->approverRepository->getApproverCategories( $this->authority->getUser()->getId() ),
```

Comment:
I suspect things will overall be easier if you replace the Authority in the constructor by `int $userId`

Reply:
This class can be renamed later, since it's no longer `AuthorityBased`.